test/prism/ruby/location_test.rb

```diff
@@ -140,6 +140,25 @@ def test_code_units
       assert_equal 7, location.end_code_units_column(Encoding::UTF_32LE)
     end
 
+    def test_code_units_handles_binary_encoding_with_multibyte_characters
+      # If the encoding is set to binary and the source contains multibyte
+      # characters, we avoid breaking the code unit offsets, but they will
+      # still be incorrect.
+
+      program = Prism.parse(<<~RUBY).value
+        # -*- encoding: binary -*-
+
+        😀 + 😀\n😍 ||= 😍
```

Comment:
A `\n` in a heredoc seems strange, I think you don't need it here, you could just end the line after the second 😀

Reply:
@vinistock I agree here, can we make this a newline?